sympy/sets/conditionset.py

```diff
@@ -33,6 +34,13 @@ class ConditionSet(Set):
     True
     """
     def __new__(cls, sym, condition, base_set):
+        if not isinstance(sym, Symbol):
+            raise TypeError('sym must be a symbol')
+        condition = as_Boolean(condition)
+        if isinstance(base_set, set):
```

Comment:
I would check that `sym` does not appear in `base_set`. Only `condition` may depend on it.

Reply:
That check happens below at line 77 after all the sym-wrangling is finished.